source/opt/cfg.cpp

```diff
@@ -58,7 +58,6 @@ void CFG::RemoveNonExistingEdges(uint32_t blk_id) {
     pred_blk->ForEachSuccessorLabel([&has_branch, blk_id](uint32_t succ) {
       if (succ == blk_id) {
         has_branch = true;
-        return;
```

Comment:
I remember requesting this.  @s-perron, I believe you said it wasn't needed.  Why, though?  Once we determine that there is a branch, we could just return from the lambda.  No point iterating over all the other successors.  Granted it's not a big saving, but I'm not sure why it's unsafe to just return once found.

Reply:
The return leaves that instance of the lambda.  It does not stop it from checking the other labels.

Reply:
If we had a "WhileEach*" version, we could return false to indicate that we should stop.